src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs

```diff
@@ -166,6 +170,131 @@ public void Dispose()
         }
 
         public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            return _concurrencyManager.Enabled ? 
+                await ExecuteWithDynamicConcurrencyAsync(cancellationToken) : 
+                await ExecuteWithConfiguredConcurrencyAsync(cancellationToken);
+        }
+
+        private async Task<TaskSeriesCommandResult> ExecuteWithDynamicConcurrencyAsync(CancellationToken cancellationToken)
+        {
+            lock (_stopWaitingTaskSourceLock)
+            {
+                if (_stopWaitingTaskSource != null)
+                {
+                    _stopWaitingTaskSource.TrySetResult(null);
+                }
+
+                _stopWaitingTaskSource = new TaskCompletionSource<object>();
+            }
+
+            // fetch another batch of messages
+            IEnumerable<CloudQueueMessage> batch = null;
+            string clientRequestId = Guid.NewGuid().ToString();
+            Stopwatch sw = null;
+            try
+            {
+                if (!_queueExists.HasValue || !_queueExists.Value)
+                {
+                    // Before querying the queue, determine if it exists. This
+                    // avoids generating unecessary exceptions (which pollute AppInsights logs)
+                    // Once we establish the queue exists, we won't do the existence
+                    // check anymore (steady state).
+                    // However the queue can always be deleted from underneath us, in which case
+                    // we need to recheck. That is handled below.
+                    _queueExists = await _queue.ExistsAsync();
+                }
+
+                if (_queueExists.Value)
+                {
+                    // determine the number of messages to pull, based on the current degree of parallelism
+                    var concurrencyStatus = _concurrencyManager.GetStatus(_functionId);
```

Comment:
Will revert the changes in this file. They show how DC can be enabled for the queues. However, we've decided to only do this for Track 2 extensions, so that extension work will happen soon in another repo.

Enabling DC for this extension is just a couple lines - mainly the call to ConcurrencyManager.GetStatus. The rest of this code is a copy of the existing listener code.

Reply:
Wait, so I've been going through this code up until this point for no reason? 😅 I guess I need to review it anyways to understand how to take advantage of it for triggers like Durable. :)